File: app/src/main/java/com/indiewalk/watchdog/earthquake/UI/MainViewModelFactory.java
package com.indiewalk.watchdog.earthquake.UI;

import android.arch.lifecycle.ViewModel;
import android.arch.lifecycle.ViewModelProvider;
import android.support.annotation.NonNull;

public class MainViewModelFactory extends ViewModelProvider.NewInstanceFactory{


        private final String listType;

        public MainViewModelFactory(String listType) {
            this.listType = listType;
        }

        @NonNull
        @Override
        public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
            return (T) new MainViewModel(listType);
        }


}

File: app/src/main/java/com/indiewalk/watchdog/earthquake/ui/MainViewModelFactory.kt
package com.indiewalk.watchdog.earthquake.ui

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider

class MainViewModelFactory(private val listType: String) : ViewModelProvider.NewInstanceFactory() {

    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        return MainViewModel(listType) as T
    }


}
